client/core/core.go

```diff
@@ -1850,10 +1870,46 @@ func (c *Core) connectAndUnlock(crypter encrypt.Crypter, wallet *xcWallet) error
 		}
 		// Notify new wallet state.
 		c.notify(newWalletStateNote(wallet.state()))
+
+		if resumeTrades {
+			go c.resumeTrades(crypter)
+		}
 	}
+
 	return nil
 }
 
+// unlockWalletResumeTrades will unlock a wallet if it is not yet unlocked.
+func (c *Core) unlockWallet(crypter encrypt.Crypter, wallet *xcWallet) error {
+	return c.unlockWalletResumeTrades(crypter, wallet, true)
+}
+
+// connectAndUnlockResumeTrades will connect to the wallet if not already
+// connected, and unlock the wallet if not already unlocked. If the wallet
+// backend is still syncing, this also starts a goroutine to monitor sync
+// status, emitting WalletStateNotes on each progress update. If resumeTrades
+// is set to true, an attempt to resume and trades that were unable to be
+// resumed at startup will be made.
+func (c *Core) connectAndUnlockResumeTrades(crypter encrypt.Crypter, wallet *xcWallet, resumeTrades bool) error {
```

Comment:
One thing that smells is the `resumeTrades bool` arguments being passed around. My intuition is that this hides an underlying deficiency where we should be doing a more sophisticated form of resolution once new wallets come online, i.e. we should be able to tell if trades need to be resumed without the caller's input.

Reply:
The reason for passing it around is to avoid recursive calls from happening when `connectAndUnlockWallet` is called from `resumeTrade`. I guess it's not completely necessary..